magneticod/magneticod/__main__.py

```diff
@@ -194,40 +216,29 @@ def parse_cmdline_arguments() -> typing.Optional[argparse.Namespace]:
         allow_abbrev=False,
         formatter_class=argparse.RawDescriptionHelpFormatter
     )
+
     parser.add_argument(
-        "--node-addr", action="store", type=str, required=False,
+        "--node-addr", action="store", type=parse_ip_port, required=False, default="0.0.0.0:0",
         help="the address of the (DHT) node magneticod will use"
     )
 
-    args = parser.parse_args(sys.argv[1:])
-
-    args.node_addr = parse_ip_port(args.node_addr) if args.node_addr else ("0.0.0.0", 0)
-    if args.node_addr is None:
-        logging.critical("Invalid node address supplied!")
-        return None
-
-    return args
-
-
-def parse_ip_port(netloc) -> typing.Optional[typing.Tuple[str, int]]:
-    # In case no port supplied
-    try:
-        return str(ipaddress.ip_address(netloc)), 0
-    except ValueError:
-        pass
+    parser.add_argument(
+        "--metadata-size-limit", type=parse_size, default=DEFAULT_MAX_METADATA_SIZE,
+        help="Limit metadata size to protect memory overflow"
```

Comment:
Also, whether this maximum size should be supplied in megabytes or kilobytes.... I guess `humanfriendly` library supports many ways of specifying the size, which is great, so we can just make it clearer.